File: py/anagrams/solution_test.py
import unittest
from solution import Solution

class TestAnagram(unittest.TestCase):
    def setUp(self):
        self.solution = Solution()

    def test_case_one(self):
        s1 = "restful"
        s2 = "fluster"
        expected = True
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

    def test_case_two(self):
        s1 = "cats"
        s2 = "tocs"
        expected = False
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

    def test_case_three(self):
        s1 = "monkeyswrite"
        s2 = "newyorktimes"
        expected = True
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

    def test_case_four(self):
        s1 = "paper"
        s2 = "reapa"
        expected = False
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

    def test_case_five(self):
        s1 = "elbow"
        s2 = "below"
        expected = True
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

    def test_case_eight(self):
        s1 = "night"
        s2 = "thing"
        expected = True
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

    def test_case_nine(self):
        s1 = "abbc"
        s2 = "aabc"
        expected = False
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

    def test_case_ten(self):
        s1 = "tax"
        s2 = "taxi"
        expected = False
        result = self.solution.anagrams(s1, s2)
        self.assertEqual(expected, result)

File: go/anagrams/solution_test.go
package solution

import "testing"

func TestAnagrams(t *testing.T) {
	testCases := []struct {
		name string
		s1   string
		s2   string
		want bool
	}{
		{"test case one", "monkeyswrite", "newyorktimes", true},
		{"test case two", "anagram", "nagaram", true},
		{"test case three", "car", "arc", true},
		{"test case four", "cats", "cots", false},
		{"test case five", "bubba", "bubbu", false},
		{"test case six", "pets", "steps", false},
		{"test case seven", "restful", "fluster", true},
		{"test case eight", "paper", "reapa", false},
		{"test case nine", "elbow", "below", true},
		{"test case ten", "night", "thing", true},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			got := anagrams(tc.s1, tc.s2)

			if got != tc.want {
				t.Errorf("got %v, want %v", got, tc.want)
			}
		})
	}
}
